torchci/lib/drciUtils.ts

```diff
@@ -338,18 +338,27 @@ export async function hasSimilarFailures(
   return false;
 }
 
-export function isInfraFlakyJob(job: RecentWorkflowsData): boolean {
-  // An infra flaky job is a failed job without any failure line and runner. It shows
-  // up as an empty job without any logs on GitHub. The failure can only be seen via
-  // the workflow summary tab
+export async function isInfraFlakyJob(
+  job: RecentWorkflowsData
+): Promise<boolean> {
+  // This covers few cases of infra flaky:
+  // * A failed job without any failure line and runner. It shows up as an empty
+  //   job without any logs on GitHub. The failure can only be seen via the workflow
+  //   summary tab
+  // * A failed job without any log on S3
+  const hasFailureLines =
+    job.failure_lines !== null &&
+    job.failure_lines !== undefined &&
+    job.failure_lines.join("") !== "";
+  const hasRunnerName =
+    job.runnerName !== null &&
+    job.runnerName !== undefined &&
+    job.runnerName !== "";
+  const hasLog = await hasS3Log(job);
```

Comment:
I feel like a better check would be to retrigger the log ingestion and then check if s3 has it?  I think I've seen a few were the log is fine on GH but I guess there was some webhook failure or something and s3 didn't have it

Reply:
That would work, log classification would be updated, but the job info would need to be re-queried to reflect the latest change.  This part makes thing complicated.  What do you think about just fire the request to ingest the log again, then leave it for the next Dr.CI request to handle the updated job info?

Reply:
I have added the function `backfillMissingLog` to re-upload the missing log to S3, this will also re-triggered the log classifier.  Testing with the missing log on https://hud.pytorch.org/pr/pytorch/pytorch/111129#18462253132

Reply:
Yup, that's how it goes